libs/parallelism/algorithms/include/hpx/parallel/algorithms/inclusive_scan.hpp

```diff
@@ -499,32 +398,496 @@ namespace hpx { namespace parallel { inline namespace v1 {
+            hpx::traits::is_iterator<FwdIter1>::value &&
+            hpx::traits::is_iterator<FwdIter2>::value
+        )>
+    // clang-format on
+    HPX_DEPRECATED_V(1, 8,
+        "hpx::parallel::inclusive_scan is deprecated, use hpx::inclusive_scan "
+        "instead")
+        typename util::detail::algorithm_result<ExPolicy, FwdIter2>::type
+        inclusive_scan(
+            ExPolicy&& policy, FwdIter1 first, FwdIter1 last, FwdIter2 dest)
+    {
+        static_assert((hpx::traits::is_forward_iterator<FwdIter1>::value),
+            "Requires at least forward iterator.");
+        static_assert((hpx::traits::is_forward_iterator<FwdIter2>::value),
+            "Requires at least forward iterator.");
 
-        return detail::inclusive_scan_(std::forward<ExPolicy>(policy), first,
-            last, dest, std::plus<value_type>(), is_segmented(),
-            util::projection_identity{});
+        using value_type = typename std::iterator_traits<FwdIter1>::value_type;
+
+#if defined(HPX_GCC_VERSION) && HPX_GCC_VERSION >= 100000
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+        return hpx::parallel::v1::detail::inclusive_scan<FwdIter2>().call(
+            std::forward<ExPolicy>(policy), first, last, dest,
+            std::plus<value_type>());
+#if defined(HPX_GCC_VERSION) && HPX_GCC_VERSION >= 100000
+#pragma GCC diagnostic pop
+#endif
     }
 }}}    // namespace hpx::parallel::v1
+
+namespace hpx {
+    ///////////////////////////////////////////////////////////////////////////
+    // DPO for hpx::inclusive_scan
+    HPX_INLINE_CONSTEXPR_VARIABLE struct inclusive_scan_t final
+      : hpx::functional::tag_fallback<inclusive_scan_t>
+    {
+        // clang-format off
+        template <typename InIter, typename OutIter,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_iterator<InIter>::value &&
+                hpx::traits::is_iterator<OutIter>::value
+            )>
+        // clang-format on
+        friend OutIter tag_fallback_dispatch(
+            hpx::inclusive_scan_t, InIter first, InIter last, OutIter dest)
+        {
+            static_assert((hpx::traits::is_input_iterator<InIter>::value),
+                "Requires at least input iterator.");
+            static_assert((hpx::traits::is_output_iterator<OutIter>::value),
+                "Requires at least output iterator.");
+
+            using value_type =
+                typename std::iterator_traits<InIter>::value_type;
+
+            return hpx::parallel::v1::detail::inclusive_scan<OutIter>().call(
+                hpx::execution::seq, first, last, dest,
+                std::plus<value_type>());
+        }
+
+        // clang-format off
+        template <typename ExPolicy, typename FwdIter1, typename FwdIter2,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::is_execution_policy<ExPolicy>::value &&
+                hpx::traits::is_forward_iterator<FwdIter1>::value &&
```

Comment:
There is now `is_forward_iterator_v<>` (same for the other iterator categories).